<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>734</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>730</width>
    <height>590</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="runButton">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>540</y>
      <width>61</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>send</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="textLine">
    <property name="geometry">
     <rect>
      <x>283</x>
      <y>541</y>
      <width>141</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser">
    <property name="geometry">
     <rect>
      <x>284</x>
      <y>483</y>
      <width>201</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="riseButton">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>490</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>rise cutter</string>
    </property>
   </widget>
   <widget class="QPushButton" name="landButton">
    <property name="geometry">
     <rect>
      <x>89</x>
      <y>490</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>land cutter</string>
    </property>
   </widget>
   <widget class="QPushButton" name="speedButton">
    <property name="geometry">
     <rect>
      <x>89</x>
      <y>530</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>set speed</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="speedBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="cutterR">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>530</y>
      <width>81</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>right cutter</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_2</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="cutterL">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>490</y>
      <width>71</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>left cutter</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_2</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="scanButton">
    <property name="geometry">
     <rect>
      <x>521</x>
      <y>490</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>scan</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBoxx">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>490</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="calibrateButton">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>520</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>calibrate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="connectButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>490</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>connect</string>
    </property>
   </widget>
   <widget class="RenderArea" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>150</y>
      <width>361</width>
      <height>301</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>300</width>
      <height>300</height>
     </size>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>381</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>moving controls</string>
    </property>
    <widget class="QSplitter" name="splitter">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>81</width>
       <height>81</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QPushButton" name="stopXButton">
      <property name="text">
       <string>stop X</string>
      </property>
     </widget>
     <widget class="QPushButton" name="parkButton">
      <property name="text">
       <string>park</string>
      </property>
     </widget>
     <widget class="QPushButton" name="stopYButton">
      <property name="text">
       <string>stop Y</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>51</width>
       <height>61</height>
      </rect>
     </property>
     <property name="title">
      <string>units</string>
     </property>
     <widget class="QCheckBox" name="mmBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>41</width>
        <height>19</height>
       </rect>
      </property>
      <property name="text">
       <string>mm</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">units</string>
      </attribute>
     </widget>
     <widget class="QCheckBox" name="stepBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>41</width>
        <height>19</height>
       </rect>
      </property>
      <property name="text">
       <string>step</string>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">units</string>
      </attribute>
     </widget>
    </widget>
    <widget class="QLineEdit" name="line">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QPushButton" name="upButton">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>20</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>↑</string>
     </property>
    </widget>
    <widget class="QPushButton" name="leftButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>←</string>
     </property>
    </widget>
    <widget class="QPushButton" name="downButton">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>↓</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rightButton">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>50</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>→</string>
     </property>
    </widget>
    <widget class="QPushButton" name="stopButton">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>50</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>□</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>170</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>2a</string>
    </property>
   </widget>
   <widget class="QComboBox" name="modeBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>mode 1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>mode 2</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>mode 3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>mode 4</string>
     </property>
    </item>
   </widget>
   <widget class="QDoubleSpinBox" name="lBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <double>70.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>230</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>L</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="aBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <double>80.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="hBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>260</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>h</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="xBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>290</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <double>14.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>290</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>x</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="yBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
    <property name="value">
     <double>22.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>320</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>y</string>
    </property>
   </widget>
   <widget class="XLine" name="xWidget" native="true">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>451</y>
      <width>361</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="YLine" name="yWidget" native="true">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>150</y>
      <width>31</width>
      <height>301</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="bBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <double>60.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>200</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>2b</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderArea</class>
   <extends>QWidget</extends>
   <header>renderarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>XLine</class>
   <extends>QWidget</extends>
   <header>xline.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>YLine</class>
   <extends>QWidget</extends>
   <header>yline.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="units"/>
  <buttongroup name="buttonGroup_2"/>
 </buttongroups>
</ui>
